compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -118,6 +118,14 @@ pub unsafe fn create_module(
     if llvm_util::get_version() < (12, 0, 0) && sess.target.arch == "powerpc64" {
         target_data_layout = strip_powerpc64_vectors(target_data_layout);
     }
+    if llvm_util::get_version() < (13, 0, 0) {
+        if sess.target.arch == "wasm32" {
+            target_data_layout = "e-m:e-p:32:32-i64:64-n32:64-S128".to_string();
+        }
+        if sess.target.arch == "wasm64" {
+            target_data_layout = "e-m:e-p:64:64-i64:64-n32:64-S128".to_string();
+        }
+    }
```

Comment:
This seems pretty poor, if we're resorting to doing this we might as well not include the data layout to the target definition at all. 

Out of scope for this PR regardless, though.

Reply:
I believe the reason the target data layout exist is that rustc needs the information itself for layout calculation and some other things.

But we don't need all the information -- only the parts parsed into TargetDataLayout. So possibly what we could do is reduce the data layout info in targets to just the necessary information, and then only check that this is consistent with the LLVM data layout (but then use the more precise LLVM data layout for code generation).

In these two cases, the newly added information is stack alignment and non-integral address spaces, both of which we don't care about.